doc/2/guides/main-concepts/3-querying/index.md

```diff
@@ -8,3 +8,601 @@ order: 300
+ - `must_not`: The clause **must not appear** in the matching documents. The score of the query will be ignored.
+
+**Example:** _Combining clauses to create an "AND" like search query_
+```bash
+kourou document:search ktm-open-data thamel-taxi '{
+  bool: {
+    filter: [
+      { term: { city: "Tirana" } },
+      { range: { age: { gte: 30 } } }
+    ]
+  }
+}'
+```
+
+**Example:** _Combining clauses to create an "OR" like search query_
+```bash
+kourou document:search ktm-open-data thamel-taxi '{
+  bool: {
+    should: [
+      { term: { city: "Siccieu" } },
+      { term: { city: "Kathmandu" } },
+    ]
+  }
+}'
+```
+
+## Sorting
+
+Elasticsearch offers the possibility to **[sort the results](https://www.elastic.co/guide/en/elasticsearch/reference/current/sort-search-results.html) by one or more fields**.
+
+**Example:** _Sort by multiple fields_
+```bash
+kourou document:search ktm-open-data thamel-taxi --sort '[
+  { city: "asc" },
+  { age: "desc" }
+]'
+```
+
+A very common sort with Kuzzle is to use [Kuzzle Metadata](/core/2/guides/main-concepts/2-data-storage#kuzzle-metadata) to sort by creation date:
+
+```bash
+kourou document:search ktm-open-data thamel-taxi --sort '[
+  { "_kuzzle_info.createdAt": "asc" }
+]'
+```
+
+Finally, if you want to always ensure the same order for your results you can sort on the `_id` field:
+
+```bash
+kourou document:search ktm-open-data thamel-taxi --sort '[
+  { "_id": "asc" }
+]'
+```
+
+## Pagination
+
+[Several pagination methods](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html) are available using Elasticsearch and Kuzzle.
+
+They allow to find all documents matching a search query.
+
+::: info
+By default, the [document:search](/core/2/api/controllers/document/search) action returns only 10 documents.  
+The number of returned documents can be changed with the `size` option.  
+There is a limit of 10000 documents that can be retrieved with a request and the limit can be configured under the `limits.documentsFetchCount` [configuration](/core/2/guides/advanced/advanced/8-configuration) key.
+:::
+
+Those methods are explained in the next sections and they are already implemented in our SDKs in the `SearchResult` class. (e.g. [SearchResult.next](/sdk/js/7/core-classes/search-result/next) method in the Javascript SDK)
+
+All the methods explained here are available by default in our SDKs through the `SearchResult` class. This class allows to navigate through your paginated results with ease by calling the `SearchResult.next` method.
+
+**Example:** _Paginate search result using the scroll method_
+:::: tabs
+
+::: Javascript
+
+```js
+let result = await sdk.document.search('ktm-open-data', 'thamel-taxi', {
+  query: {
+    term: { city: 'Antalya' }
+  }
+},
+{ scroll: '5s' });
+
+while (result) {
+  console.log(result.hits);
+  result = await result.next();
+}
+```
+
+:::
+::: Dart
+
+
+```dart
+var result = await kuzzle
```

Comment:
@jenow @Yoann-Abbes is it correct?

Reply:
I added corrections
You can see an example of search with scroll here
https://docs.kuzzle.io/sdk/dart/2/core-classes/search-result/next/